engine/crates/sdl-parser/src/lib.rs

```diff
@@ -144,8 +165,79 @@ pub async fn parse<'a>(
         directives.to_definition(),
     );
 
-    let schema = parse_schema(schema)?;
+    grafbase_engine::parser::parse_schema(schema)
+}
 
+pub fn parse_used_directives(schema: &str) -> grafbase_engine::parser::Result<UsedDirectives> {
+    let doc = parse_schema(schema)?;
+    let mut used = UsedDirectives::default();
+    fn directive_name(directive: Positioned<ConstDirective>) -> String {
+        directive.node.name.node.to_string()
+    }
+    for definition in doc.definitions {
+        match definition {
+            TypeSystemDefinition::Schema(schema) => used
+                .schema
+                .extend(schema.node.directives.into_iter().map(directive_name)),
+            TypeSystemDefinition::Type(ty) => {
+                used.r#type.extend(ty.node.directives.into_iter().map(directive_name));
+                match ty.node.kind {
+                    TypeKind::Object(object) => {
+                        for field in object.fields {
+                            used.object_field
+                                .extend(field.node.directives.into_iter().map(directive_name));
+                            used.input_value.extend(
+                                field
+                                    .node
+                                    .arguments
+                                    .into_iter()
+                                    .flat_map(|arg| arg.node.directives)
+                                    .map(directive_name),
+                            );
+                        }
+                    }
+                    TypeKind::Interface(interface) => {
+                        for field in interface.fields {
+                            used.interface_field
+                                .extend(field.node.directives.into_iter().map(directive_name));
+                            used.input_value.extend(
+                                field
+                                    .node
+                                    .arguments
+                                    .into_iter()
+                                    .flat_map(|arg| arg.node.directives)
+                                    .map(directive_name),
+                            );
+                        }
+                    }
+                    TypeKind::Enum(r#enum) => {
+                        for value in r#enum.values {
+                            used.enum_value
+                                .extend(value.node.directives.into_iter().map(directive_name));
+                        }
+                    }
+                    TypeKind::InputObject(input_object) => {
+                        for field in input_object.fields {
+                            used.input_value
+                                .extend(field.node.directives.into_iter().map(directive_name));
+                        }
+                    }
+                    TypeKind::Union(_) | TypeKind::Scalar => (),
```

Comment:
Should we also gather directives from the enum definition, input object definition and the unions & scalars?

Not super important since we don't have any directives on those atm, but would save us having to remember to update this in the future.

Reply:
Oh wait, never mind - I see those are being gathered outside of this case.

Reply:
damn, you're right missed it. Those aren't defined by the type kind but by the type definition. Thanks!

Reply:
oh yes, right xD